bugwarrior/services/__init__.py

```diff
@@ -481,22 +481,21 @@ def __repr__(self):
         return '<%s>' % self.__unicode__()
 
 
-class ServiceClient(object):
-    """ Abstract class responsible for making requests to service API's. """
-    @staticmethod
-    def json_response(response):
-        # If we didn't get good results, just bail.
-        if response.status_code != 200:
-            raise IOError(
-                "Non-200 status code %r; %r; %r" % (
-                    response.status_code, response.url, response.text,
-                ))
-        if callable(response.json):
-            # Newer python-requests
-            return response.json()
-        else:
-            # Older python-requests
-            return response.json
+def json_response(response):
+    """ Process a request's response to keep HTTP error handling consistent
+    accross services. """
+    # If we didn't get good results, just bail.
+    if response.status_code != 200:
+        raise IOError(
+            "Non-200 status code %r; %r; %r" % (
+                response.status_code, response.url, response.text,
+            ))
+    if callable(response.json):
+        # Newer python-requests
+        return response.json()
+    else:
+        # Older python-requests
+        return response.json
```

Comment:
> a common pattern doesn't necessarily imply shared code and shared code doesn't necessarily imply a common parent class.

I agree with this, which is why I'm now emphasizing the communicative value `ServiceClient` rather than code sharing.

I would like to see further standardization of the service structure and interfaces. The more standardized they are, the less effort it is to jump into a new one.

Python's minimalist design means that classes serve multiple valid purposes. The most straightforward way to express that a class serves a common purpose or interface is a base class, even if that base class doesn't actually do anything.

Several of the services have `ServiceClient`'s whether you want to give them a base class or not. If `ServiceClient` is a good pattern, I think it should have a base class. If it's a bad pattern, we should purge them and roll them into `IssueService`.

> But I can live with it either way, so if you feel strongly about it tell me and I'll revert the commit.

I'd prefer reverting and giving `ServiceClient` a bit more time to prove itself a useful abstraction. Anyone else should feel free to weigh in.